### PR TITLE
internal/plugins/helm: scaffold finalizers subresource permission for created APIs

### LINKED ISSUE #3767: Quickstart for Helm-based Operators is not working on OpenShift
## Bug Report

**What did you do?**
I followed https://master.sdk.operatorframework.io/docs/building-operators/helm/quickstart/

```
mkdir nginx-operator
cd nginx-operator
operator-sdk init --plugins=helm
operator-sdk create api --group demo --version v1 --kind Nginx
make docker-build docker-push IMG=docker.io/nmasse/nginx-operator:test
make install
make deploy IMG=docker.io/nmasse/nginx-operator:test
kubectl apply -f config/samples/demo_v1_nginx.yaml
kubectl get all -l "app.kubernetes.io/instance=nginx-sample"
```

**What did you expect to see?**

I expected to see an nginx pod come up. But it never came up.

**What did you see instead? Under which circumstances?**

The operator logs exhibit an error `failed to install release: serviceaccounts \"nginx-sample\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on`

Full log file:

```
{"level":"info","ts":1597936277.6508987,"logger":"cmd","msg":"Version","Go Version":"go1.13.15","GOOS":"linux","GOARCH":"amd64","helm-operator":"v1.0.0"}
{"level":"info","ts":1597936277.6517258,"logger":"cmd","msg":"WATCH_NAMESPACE environment variable not set. Watching all namespaces.","Namespace":""}
{"level":"info","ts":1597936281.6490111,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"127.0.0.1:8080"}
{"level":"info","ts":1597936281.6504083,"logger":"helm.controller","msg":"Watching resource","apiVersion":"demo.my.domain/v1","kind":"Nginx","namespace":"","reconcilePeriod":"1m0s"}
I0820 15:11:21.651919       1 leaderelection.go:242] attempting to acquire leader lease  nginx-operator-system/nginx-operator...
{"level":"info","ts":1597936281.7339883,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
I0820 15:11:39.625672       1 leaderelection.go:252] successfully acquired lease nginx-operator-system/nginx-operator
{"level":"info","ts":1597936299.6272435,"logger":"controller","msg":"Starting EventSource","controller":"nginx-controller","source":"kind source: demo.my.domain/v1, Kind=Nginx"}
{"level":"info","ts":1597936299.7290106,"logger":"controller","msg":"Starting Controller","controller":"nginx-controller"}
{"level":"info","ts":1597936299.7297232,"logger":"controller","msg":"Starting workers","controller":"nginx-controller","worker count":4}
{"level":"error","ts":1597936394.6697314,"logger":"helm.controller","msg":"Release failed","namespace":"nginx-operator","name":"nginx-sample","apiVersion":"demo.my.domain/v1","kind":"Nginx","release":"nginx-sample","error":"failed to install release: serviceaccounts \"nginx-sample\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/operator-framework/operator-sdk/internal/helm/controller.HelmOperatorReconciler.Reconcile\n\tsrc/github.com/operator-framework/operator-sdk/internal/helm/controller/reconcile.go:181\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:90"}
{"level":"error","ts":1597936394.6907682,"logger":"controller","msg":"Reconciler error","controller":"nginx-controller","name":"nginx-sample","namespace":"nginx-operator","error":"failed to install release: serviceaccounts \"nginx-sample\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:237\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:90"}
```

**Environment**
* operator-sdk version:

```
operator-sdk version: "v1.0.0", commit: "d7d5e0cd6cf5468bb66e0849f08fda5bf557f4fa", kubernetes version: "v1.18.2", go version: "go1.13.11 darwin/amd64", GOOS: "darwin", GOARCH: "amd64"
helm-operator version: "v1.0.0", commit: "d7d5e0cd6cf5468bb66e0849f08fda5bf557f4fa", kubernetes version: "v1.18.2", go version: "go1.13.11 darwin/amd64", GOOS: "darwin", GOARCH: "amd64"
```

* Kubernetes version information:

I'm using OpenShift Version 4.3.5, which is based on Kubernetes v1.16.2. 
I'm the cluster-admin of this cluster.

* Kubernetes cluster kind: 3 masters, 3 workers, in VM. 

* Are you writing your operator in ansible, helm, or go?

Helm

**Possible Solution**

The generated `config/rbac/role.yaml` is missing privileges on `nginxes/finalizers`. 
The `operator-sdk create api` command that generates this file needs to be updated to include this privilege since it is required on OpenShift. 

```yaml
##
## Rules for demo.my.domain/v1, Kind: Nginx
##
- apiGroups:
  - demo.my.domain
  resources:
  - nginxes
  - nginxes/status
  - nginxes/finalizers
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
```

Since the generated objects include a `ownerReference` field that references the created CR, on OpenShift the operator needs to be able to set finalizers on the created CR. 

```
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    meta.helm.sh/release-name: nginx-sample
    meta.helm.sh/release-namespace: nginx-operator
  creationTimestamp: "2020-08-20T15:19:08Z"
  labels:
    app.kubernetes.io/instance: nginx-sample
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: nginx
    app.kubernetes.io/version: 1.16.0
    helm.sh/chart: nginx-0.1.0
  name: nginx-sample
  namespace: nginx-operator
  ownerReferences:
  - apiVersion: demo.my.domain/v1
    blockOwnerDeletion: true
    controller: true
    kind: Nginx
    name: nginx-sample
    uid: c5d7e996-013f-40ca-bd19-14ba73728eaf
  resourceVersion: "150618365"
  selfLink: /api/v1/namespaces/nginx-operator/serviceaccounts/nginx-sample
  uid: ce916607-58dd-4c1f-bf1e-dbf6f6e86c04
secrets:
- name: nginx-sample-token-99vgm
- name: nginx-sample-dockercfg-rrzb7
```

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**

Add `<resource>/finalizers` subresource RBAC permission for Helm operator APIs

**Motivation for the change:**

Closes #3767

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,7 +35,7 @@
 - When generating bundles and packagemanifests, remove `metadata.namespace` from namespaced resources when writing them into the `manifests` directory to avoid validation errors. ([#3813](https://github.com/operator-framework/operator-sdk/pull/3813))
 - Fixed a bug that caused the Helm operator not to set the `InstallSuccessful` and `UpgradeSuccessful` status reasons when the status update fails during installation and upgrade. ([#3735](https://github.com/operator-framework/operator-sdk/pull/3735))
 - Bumped helm and k8s dependencies to v3.3.4 and v1.18.8 to fix [this upstream bug](https://github.com/kubernetes/kubernetes/issues/91615). ([#3936](https://github.com/operator-framework/operator-sdk/pull/3936))
-- In Helm projects, fix operator permissions for Openshift deployments by adding a `<resource>/finalizers` rule in the operator's role. ([#3779](https://github.com/operator-framework/operator-sdk/pull/3779))
+- In Ansible projects, fix operator permissions for Openshift deployments by adding a `<resource>/finalizers` rule in the operator's role. ([#3779](https://github.com/operator-framework/operator-sdk/pull/3779))
```
Comment (estroz) [inline CHANGELOG.md:38]: Can you also correct this in the published release notes?

Comment (joelanford) [reply CHANGELOG.md:38]: I'll do that when this merges.

**Review by camilamacedo86 — APPROVED** [review]
/approve 
/lgtm

**Review by estroz — APPROVED** [review]
/lgtm

File: CHANGELOG.md (hunk 1 above)
Comment (joelanford) [reply CHANGELOG.md:38]: Done: Updated https://github.com/operator-framework/operator-sdk/releases/tag/v1.1.0

**Pull request merged**